pyccel/parser/syntactic.py

```diff
@@ -643,6 +647,21 @@ def fill_types(ls):
             return container
 
         decorators = {}
+
+        # add the decorator @types if the arguments are annotated
+        annotated_args = []
+        for a in arguments:
+            if isinstance(a, Argument):
+                annotated_args.append(a.annotation)
+            elif isinstance(a, ValuedArgument):
+                annotated_args.append(a.argument.annotation)
```

Comment:
Would this not be faster with list comprehensions. Ie:
```
annotated_args = [a.annotation if isinstance(a, Argument) else a.argument.annotation \
                  for a in arguments if isinstance(a, (Argument, ValuedArgument)]
```
or perhaps better:
```
annotated_args  = [a.annotation for a in arguments if isinstance(a, Argument)]
annotated_args += [a.argument.annotation for a in arguments if isinstance(a, ValuedArgument)]
```

Reply:
I didn't do it, because it becomes unreadable

Reply:
Actually I find list comprehensions very readable in general, but maybe not much so in this case.

The second suggestion of @EmilyBourne changes the order of the elements in `annotated_args` compared to the original code of @saidctb. Is that a problem?

Reply:
yes we need to keep the order